hazelcast/src/main/java/com/hazelcast/internal/cluster/impl/ClusterHeartbeatManager.java

```diff
@@ -644,9 +674,91 @@ private void resetHeartbeats() {
     /** Remove the {@code member}'s heartbeat timestamps */
     void removeMember(MemberImpl member) {
         heartbeatFailureDetector.remove(member);
+        if (icmpParallelMode) {
+            icmpFailureDetector.remove(member);
+        }
     }
 
     void reset() {
         heartbeatFailureDetector.reset();
+        if (icmpParallelMode) {
+            icmpFailureDetector.reset();
+        }
+    }
+
+    private class PingTask
+            implements Runnable {
+
+        final Member member;
+
+        PingTask(Member member) {
+            this.member = member;
+        }
+
+        public void run() {
+            try {
+                Address address = member.getAddress();
+                logger.warning(format("%s will ping %s", node.getThisAddress(), address));
+                for (int i = 0; i < MAX_PING_RETRY_COUNT; i++) {
+                    if (doPing(address)) {
+                        return;
+                    }
+                }
+                // host not reachable
+                String reason = format("%s could not ping %s", node.getThisAddress(), address);
+                logger.warning(reason);
+                clusterService.suspectMember(member, reason, true);
+            } catch (Throwable ignored) {
+                EmptyStatement.ignore(ignored);
+            }
+        }
+
+        boolean doPing(Address address)
+                throws IOException {
+            try {
+                if (address.getInetAddress().isReachable(null, icmpTtl, icmpTimeoutMillis)) {
+                    logger.info(format("%s pinged %s successfully", node.getThisAddress(), address));
```

Comment:
I believe this should not be an INFO level, it's flooding the member's output if working properly.